src/python/pants/backend/jvm/subsystems/jvm_platform.py

```diff
@@ -15,7 +16,35 @@
 
 
 class JvmPlatform(Subsystem):
-  """Used to keep track of repo compile and runtime settings for jvm targets."""
+  """Used to keep track of repo compile and runtime settings for jvm targets.
+
+  JvmPlatform covers both compile time and runtime jvm platform settings. A platform is a group of
+  compile time and runtime configurations.
+
+  For compile time configuration, there are three main attributes, `source`, `target` and `args`.
+  `source` and `target` map directly to the javac arguments of the same name, though they accept
+   more aliases than javac. `args`, allows the platform to specify additional, global to that
+   platform, compile arguments.
+
+  For runtime configuration, `target` and `jvm_options` are used. `target` is used to determine the
+  version of the JVM to use. `jvm_options` is used by some tasks to allow platforms to specify
+  platform specific JVM options. These can make transitions between JVM versions smoother by
+  allowing compatibility options to be provided globally.
+
+  An example pants.toml config might look like this:
+
+      [jvm-platform]
+      default_platform: java8
+      default_runtime_platform: java10
+      platforms =
+      \"\"\"
+       {
+          'java8': { 'source': '8', 'target': '8', 'args': [] },
+          'java10': {'source': '10', 'target': '10', 'args': [],
+                     'jvm_options': ['-Djava.compat.foo=act-like-java-8'] },
+        }
+     \"\"\"
+  """
```

Comment:
Great documentation! Could we add it to `src/docs/common_tasks/jvm_options.md`?

Reply:
I considered that, but that particular doc is focused on passing jvm-options at use, which is a slightly different use case. The platform jvm options docs would need to be more targeted for repo owners rather than users. Is there a good place for that?

Reply:
I understand your distinction, but I'd prefer to make this information as discoverable as possible, and as that page is just titled "Specify JVM Options" I think it should be sufficiently appropriate. That page is published to the website at just https://www.pantsbuild.org/jvm_options.html, without any prefix, so while the other "common tasks" have a very fixed template, I think we should be able to just make separate sections for "JVM Options at Runtime" vs "Setting JVM Options for a Repo".

Reply:
Works for me. I'll move it there and add a reference.